lib/utils.js

```diff
@@ -68,45 +43,57 @@ const getDataStats = (data, checksum, dedupHash) => ({
   size: Buffer.byteLength(data),
 });
 
-const compress = (file, minCompressSize, compression, cb) => {
-  fs.stat(file, (err, stats) => {
-    if (err || stats.size <= minCompressSize) {
-      cb(err, false);
-      return;
-    }
-    const filec = file + 'z';
-    const compressor = compressors[compression];
-    if (!compressor) {
-      throw new Error(`Unknown compression type ${compression} specified`);
-    }
-    compressor
-      .compressFile(file, filec)
-      .then(() =>
-        fs.rename(filec, file, err => {
-          if (err) cb(err, false);
-          else cb(null, true);
-        })
-      )
-      .catch(err => cb(err, false));
-  });
+const mkdirRecursive = util.promisify(common.mkdirp);
+
+const rmRecursive = async dir => {
+  try {
+    await fs.access(dir);
+  } catch (err) {
+    if (err.code === 'ENOENT') return;
+    throw new Error(`Cannot access directory '${dir}': ${err.message}`);
+  }
+
+  const files = await fs.readdir(dir);
+
+  for (let file of files) {
+    file = path.join(dir, file);
+    const stats = await fs.stat(file);
+
+    if (stats.isDirectory()) await rmRecursive(file);
+    else await fs.unlink(file);
+  }
+
+  await fs.rmdir(dir);
 };
 
-const uncompress = (file, opts, cb) => {
-  fs.access(file, err => {
-    if (err) {
-      cb(err);
-      return;
-    }
-    const compressor = compressors[opts.compression];
-    if (!compressor) {
-      throw new Error(`Unknown compression type ${opts.compression} specified`);
-    }
+const compress = async (file, minCompressSize, compression) => {
+  const compressor = compressors[compression];
+  if (!compressor) {
+    throw new Error(`Unknown compression type ${compression} specified`);
+  }
+
+  const stats = await fs.stat(file);
+  if (stats.size <= minCompressSize) return false;
```

Comment:
Unrelated to this PR, but:

Now that I think of it, wouldn't this lead for the user of Filestorage to a need to keep a map of compressed/uncompressed files? Because without such map if someone calls `compress(file, 100, 'xz')` and the file size is less than 100 then the file will not be compressed and if the user then calls `read(file, { compression: 'xz' })` this would lead to an error as the file was not compressed?

Reply:
The idea was that metadata including file name, hash sum, deduplication hash, compression type etc. will be stored in the DB. I don't see a problem with the need for such a map since you already have to store files compression types somewhere.